### ISSUE TITLE
Analysis and/or disassembly of windows binaries has serious problems

### ISSUE THREAD

**Issue by newlog** [radare/radare2#6033]
There's a problem with the analysis done by new versions of radare2 in some windows binaries. 

Case: Practical Malware Analysis, Lab07-01.exe.

With older r2 versions, the analysis of this binary worked fine. Instruction at `0x0040123f` was a call to the main function, whereas now it shows `call sub.ADVAPI32.dll_StartServiceCtrlDispatcherA_0`.

```
|           0x00401228      a114534000     mov eax, dword [0x405314]   ; [0x405314:4]=0
|           0x0040122d      a318534000     mov dword [0x405318], eax   ; [0x405318:4]=0
|           0x00401232      50             push eax
|           0x00401233      ff350c534000   push dword [0x40530c]
|           0x00401239      ff3508534000   push dword [0x405308]
|           0x0040123f      e8bcfdffff     call sub.ADVAPI32.dll_StartServiceCtrlDispatcherA_0
|           0x00401244      83c40c         add esp, 0xc
```

The problem is not just with the call, but previous instructions disassembly is also different.

With the new behavior, when following the call to `sub.ADVAPI32.dll_StartServiceCtrlDispatcherA_0`, the code inside is the expected (and not the advapi32 code). In fact, inside this call there's a call to `call dword [sym.imp.ADVAPI32.dll_StartServiceCtrlDispatcherA]`. This seems like a common pattern in all the places where this issue happens.

If you look into the `sub.ADVAPI32.dll_StartServiceCtrlDispatcherA_0` function, the same problem happens:

```
|           0x00401028      ff1504404000   call dword [sym.imp.ADVAPI32.dll_StartServiceCtrlDispatcherA] ;[1] ; "&F" @ 0x404004                                                 
|           0x0040102e      6a00           push 0                                                                                                                               
|           0x00401030      6a00           push 0                                                                                                                               
|           0x00401032      e809000000     call sub.KERNEL32.dll_OpenMutexA_40 ;[2]                                                                                             
|           0x00401037      83c418         add esp, 0x18                                                                                                                        
=<           0x0040103a      c3             ret
```

With old r2 behavior, it showed:

![7-1 1 1](https://cloud.githubusercontent.com/assets/950789/19701693/c9343ae8-9afc-11e6-9798-595bd85bb5fe.png)

The call to `sub.KERNEL32.dll_OpenMutexA_40` should be a call to fcn.0401040 as what happened with old r2. In this case, inside this function the same happens, the function `call dword [sym.imp.KERNEL32.dll_OpenMutexA]` is called.

Long story short, at some point, analysis got fucked up. This happens consistently in many places of the binary. The binary does not contain any type of anti-re feature so this happens.

**Comment by Maijin** [comment]
This is due to autorename, if you don't like it, don't use it.

**Comment by Maijin** [comment]
e anal.autoname = false

**Comment by Maijin** [comment]
You can also help improving it

**Comment by newlog** [comment]
I don't think it's a matter of taste. Isn't it that it's wrong?

In any case, it seems that autoname=false does not fix it: http://asciinema.org/a/ane7y73almr0gvdn9fy31bvmb

**Comment by Maijin** [comment]
You have to run `aaa` after `e anal.autoname=false`

**Comment by Maijin** [comment]
And it is definitely matter of taste I got many many binaries which without autorename are making r2 completely useless.

**Comment by alvarofe** [comment]
definitely isn't wrong. just configure the variable as you wish in `.radarer2c` or otherwise don't throw -AAAAAAAAAAAA, set the eval before to run any analysis in the prompt

**Comment by alvarofe** [comment]
tbh i find it very useful

**Comment by Maijin** [comment]
It's very useful but would definitely need real improvement.

**Comment by newlog** [comment]
Oh damn! I thought the more "A" the more powerful r2 was! :(

Ok, so for e anal.autoname=false` to work after the `aaa`, you have to execute r2 without the `-A` in the beginning.

I guess I'll have to understand the logic behind calling a user function with the name of a Windows API function. Maybe the trailing number in the WinAPI function name has a meaning. For me it was misguiding having a user function with the name of a Windows API (why?), but I guess I should have got it when seeing the prefix "sub" instead of "sym.imp". 

Good to know, if that's the expected behavior (beware, my dear n00bs -like myself-!) feel free to close it!

**Comment by newlog** [comment]
Just for completeness, the function is shown with that name because that's the first symbol that will be called inside of the function itself. Instead of showing a random `fcn` name, internal functions are renamed with this approach.
